juriscraper/opinions/united_states/federal_bankruptcy/bap1.py

```diff
@@ -0,0 +1,175 @@
+"""
+Scraper for the United States Bankruptcy Appellate Panel for the First Circuit
+CourtID: bap1
+Court Short Name: 1st Cir. BAP
+Court Contact: ca01_BAP@ca1.uscourts.gov, (617) 748-9650
+Author: Gianfranco Rossi
+History:
+ - 2023-12-28, grossir: created
+"""
+import calendar
+import re
+from typing import Any, Dict, Optional
+
+from lxml.html import HtmlElement
+
+from juriscraper.lib.exceptions import SkipRowError
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
+
+
+class Site(OpinionSiteLinear):
+    lower_court_to_abbreviation = {
+        "USBC - District of New Hampshire": "NH",
+        "USBC - District of Massachusetts (Worcester)": "MW",
+        "USBC - District of Puerto Rico": "PR",
+        "USBC - District of Massachusetts (Boston)": "MB",
+        "USBC - District of Maine (Portland)": "EP",
+        "USBC - District of Rhode Island": "RI",
+        "Bankrupcty Court of ME - Bangor": "EB",
+        "Bankruptcy Court of MA - Boston": "MB",
+        "Bankruptcy Court of MA - Springfield": "MS",
+        "Bankruptcy Court of ME - Portland": "EP",
+        "Bankruptcy Court - Rhode Island": "RI",
+        "Bankruptcy Court - San Juan Puerto Rico": "PR",
+        "Bankruptcy Court of MA - Worcester": "MW",
+        "Bankruptcy Court - Ponce Puerto Rico": "PR",
+        "Bankruptcy Court of NH, Concord": "NH",
+    }
+    # See issue #828 for mapper extraction reference
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.base_url = self.url = "https://www.bap1.uscourts.gov/bapopn"
+        self.court_id = self.__module__
+
+        # There are 29 historical pages as of development in Dec 2023
+        # source indexes from 0
+        self.back_scrape_iterable = range(29)[::-1]
+
+    def _download(self, request_dict: Optional[Dict] = None) -> HtmlElement:
+        """
+        Gets the source's HTML
+
+        :param request_dict: unused in this scraper
+        :return: HtmlElement object
```

Comment:
same thing tiny tweaks to docstring - and I wouldnt but the type here for return.  

we know its type from the type hints.  you could say - the html element of the downloaded page?  if you wanted